### PR TITLE
All listeners and cookie replay

### PR DESCRIPTION
**Pull request by Toflar**
Todo:

- [x] Unit tests

### CODE REVIEW

File: src/EventListener/HeaderReplayStopPropagationListener.php [hunk 1]
```diff
@@ -18,40 +18,30 @@
 /**
  * Class HeaderReplayStopPropagationListener.
  *
- * This listener is registered with a very high priority on both, the
- * kernel.response and the kernel.terminate listeners. It makes sure
- * it stops propagation of these events if the current request is a preflight
- * request. It never makes sense to modify a preflight request by one of these
- * events as the bundle provides its own event for this.
+ * This listener is registered with a very high priority on the kernel.terminate
+ * listeners. It makes sure it stops propagation of these events if the current
```
Comment (leofeyer) [inline src/EventListener/HeaderReplayStopPropagationListener.php]: `events. It makes sure to stop propagation of the event if the current`

File: src/EventListener/HeaderReplayStopPropagationListener.php [hunk 2]
```diff
@@ -18,40 +18,30 @@
...
-    /**
-     * @param KernelEvent $event
-     */
-    private function stopPropagationIfApplicable(KernelEvent $event)
-    {
-        $request = $event->getRequest();
-
-        if (in_array(HeaderReplayListener::CONTENT_TYPE, $request->getAcceptableContentTypes(), true)) {
+        if (in_array(
+            HeaderReplayListener::CONTENT_TYPE,
```
Comment (leofeyer) [inline src/EventListener/HeaderReplayStopPropagationListener.php:41]: But Varnish does not send the `application/vnd.t42.header-replay` content type, does it?

File: src/SymfonyCache/HeaderReplaySubscriber.php [hunk 3]
```diff
@@ -108,8 +110,17 @@ public function preHandle(CacheEvent $event)
             $headersToReplay = explode(',', $preflightResponse->headers->get(HeaderReplayListener::REPLAY_HEADER_NAME));
 
             foreach ($headersToReplay as $header) {
+
+                // Make sure Set-Cookie is never considered because we do this
+                // manually
+                $header = strtolower($header);
+                unset($headersToReplay['set-cookie']);
```
Comment (leofeyer) [inline src/SymfonyCache/HeaderReplaySubscriber.php]: Should `unset()` not be called before the loop?

File: src/EventListener/HeaderReplayStopPropagationListener.php (hunk 2 above)
Comment (Toflar) [reply src/EventListener/HeaderReplayStopPropagationListener.php:41]: It has to.

File: src/EventListener/HeaderReplayStopPropagationListener.php (hunk 1 above)
Comment (Toflar) [reply src/EventListener/HeaderReplayStopPropagationListener.php]: Updated the docs.

File: src/SymfonyCache/HeaderReplaySubscriber.php (hunk 3 above)
Comment (Toflar) [reply src/SymfonyCache/HeaderReplaySubscriber.php]: Ah, that was an issue I would've found with unit tests anyway :)

File: src/EventListener/HeaderReplayStopPropagationListener.php (hunk 2 above)
Comment (leofeyer) [reply src/EventListener/HeaderReplayStopPropagationListener.php:41]: So how do I configure Varnish then?

Comment (Toflar) [reply src/EventListener/HeaderReplayStopPropagationListener.php:41]: https://github.com/terminal42/header-replay-bundle#varnish

**Comment by leofeyer** [comment]
The changes work very well and also fix another issue we had (the user was only authentication via remember me upon the second request). So thumb up and thank you!

**Pull request merged**